### PR TITLE
test(launchpad): add e2e tests for Project Setup - Round 3

### PR DESCRIPTION
**Pull request by emilyrohrbough**
- Apart of #19051

**Test Cases**
- component Project Setup (never having setup CT before)
    - [x] On the 'choose framework' page, clicking 'back' returns you to the 'project setup' page to choose e2e or component
    - [x] The framework dropdown displays all frameworks supported by Cypress with their icon  (Create React App, Vue CLI, React.js, Vue.js, Next.js, Nuxt.js)
    - [x] The Bundler dropdown display all bundlers supported by Cypress with their icon (Webpack, Vite)
    - [x] Choosing each option on the 'choose framework' page shows the items in the dropdowns as selected and allows you to select the 'Next' button
    - [x] Clicking Next after selecting options on the 'choose framework' page takes you to the 'install dev dependency' screen.

- component Install Dev Dependencies
  - [x] After selecting framework, bundler, from previous project setup, 'install dev dependencies' page is displayed
  - [x] Clicking 'Back' on the 'install dev dependencies' page returns you to the 'choose framework' page
  - [x] The 'install dev dep' page has an area that displays the command to run to install the required packages
  - [x] Clicking 'copy' next to the command gives a visual indicator that the command was copied
  - [x] Clicking 'copy' next to the command copies the command to the clipboard
  - [x] Below the command, there are n rows shown for each package displaying the package name and a description on what the package does
  - [x] Clicking 'Continue' routes you to the 'configured files' page.

**TO DO:**
- Update missing requirement(s) around storybook detection & add tests

### CODE REVIEW

**Comment by ZachJW34** [comment]
I still get stuck in the Framework/bundler without being able to navigate forward if I:
* Choose a project that has not been setup -> Click component testing
* Choose "React.js" for framework, select "Webpack" or "Vite" for Bundler
* Choose "Create React App" from framework
 The "Next Step" becomes disabled even though the options are valid.
 
Also, if you choose "Create React App" and then refresh the page, it will break.


https://user-images.githubusercontent.com/25158820/150438562-bd907ae7-5e6c-422b-ab70-39d19708f1cc.mov

File: packages/launchpad/cypress/e2e/project-setup.cy.ts [hunk 1]
```diff
@@ -201,33 +217,58 @@ describe('Launchpad: Setup Project', () => {
...
+    describe('project not been configured for cypress', () => {
+      it('can setup e2e testing for a project', () => {
+        cy.openProject('pristine')
+        cy.visitLaunchpad()
 
-      verifyWelcomePage({ e2eIsConfigured: false, ctIsConfigured: false })
+        verifyWelcomePage({ e2eIsConfigured: false, ctIsConfigured: false })
+
+        // @ts-ignore
+        cy.get('body').tab()
```
Comment (tbiethman) [inline packages/launchpad/cypress/e2e/project-setup.cy.ts:228]: Should this be using realPress like above?

File: packages/launchpad/cypress/e2e/project-setup.cy.ts [hunk 2]
```diff
@@ -257,6 +298,151 @@ describe('Launchpad: Setup Project', () => {
...
+
+      const hasStorybookPermutations = [false, true]
+
+      FRONTEND_FRAMEWORKS.forEach((framework) => {
+        hasStorybookPermutations.forEach((hasStorybookDep) => {
+          framework.supportedBundlers.forEach((testBundler) => {
+            const bundler = BUNDLERS.find((b) => b.type === testBundler)
+
+            if (!bundler) {
+              throw new Error(`${framework.name} claims to support the bundler, ${testBundler}, however it is not a valid Cypress bundlers.`)
```
Comment (tbiethman) [inline packages/launchpad/cypress/e2e/project-setup.cy.ts:323]:
```suggestion
              throw new Error(`${framework.name} claims to support the bundler, ${testBundler}, however it is not a valid Cypress bundler.`)
```

**Comment by emilyrohrbough** [comment]
@ZachJW34 Nice catch. I made some changes to how the components are managing how it's updated / rendered to keep the UI in sync with the data context. I know it's been discussed keeping track of UI states instead of relying on the GQL update but in this case, the data context will update more than one property [here](https://github.com/cypress-io/cypress/blob/10.0-release/packages/data-context/src/actions/WizardActions.ts#L33), which was getting lost on the UI.

**Review by tbiethman — DISMISSED** [review]
Nice job, this adds a bunch of coverage very efficiently :+1:

**Review by ZachJW34 — APPROVED** [review]

File: packages/launchpad/tsconfig.json [hunk 3]
```diff
@@ -11,6 +11,7 @@
     "../frontend-shared/cypress/**/*.ts"
   ],
   "compilerOptions": {
+    "types": ["cypress", "cypress-real-events"],
```
Comment (ZachJW34) [inline packages/launchpad/tsconfig.json:14]: Adding `"@intlify/vite-plugin-vue-i18n/client"` fixes the `checkTs` failure.

**Review by ZachJW34 — APPROVED** [review]

**Review by tbiethman — APPROVED** [review]

**Review by tbiethman — APPROVED** [review]

**Pull request merged**